components/mailbox/src/Mailbox.svelte

```diff
@@ -351,11 +375,19 @@
         (thread) => thread !== openedEmailData,
       );
       starredThreads.delete(openedEmailData);
+      unreadThreads.delete(openedEmailData);
       selectedThreads.delete(openedEmailData);
       openedEmailData = null;
     } else {
-      selectedThreads.forEach((thread) => {
+      selectedThreads.forEach(async (thread) => {
         starredThreads.delete(thread);
+        unreadThreads.delete(thread);
+        if (trashLabelID) {
+          thread.label_ids = [trashLabelID];
+        } else if (trashFolderID) {
+          thread.folder_id = trashFolderID;
+        }
```

Comment:
Just a consideration here. If we somehow never got a valid folder or label, this action does nothing effectively - but we do show in the UI that it has been "deleted".  Maybe we should handle that scenario?

Reply:
I wonder when would one run into that situation. May be when some deletes their `Trash` folder/label (Not entirely sure if that's possible)?
But you're right, we should handle this better. May be a good way to handle it would be to not update the UI and raise a debug error instead. Wdyt?